src/test/java/io/github/oliviercailloux/y2018/apartments/valuefunction/ApartmentValueFunctionTest.java

```diff
@@ -46,25 +49,113 @@ void apartmentValueFunctionTest() throws NumberFormatException {
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setNbBathroomsSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setTerraceSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setFloorAreaTerraceSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setWifiSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setPricePerNightSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setNbMinNightSubjectiveValueWeight(-1d));
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.setTeleSubjectiveValueWeight(-1d));
+
+	}
+
+	/**
+	 * Function to test the adaptation to the subjective value weight of a criteria
+	 */
+	@Test
+	void adaptWeightTest() {
+
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.adaptWeight(Criterion.TELE, Criterion.TELE));
+		valueFunction.setTeleSubjectiveValueWeight(7d);
+		valueFunction.setTerraceSubjectiveValueWeight(3d);
+		assertEquals(7d, valueFunction.getSubjectiveValueWeight(Criterion.TELE));
+		assertEquals(3d, valueFunction.getSubjectiveValueWeight(Criterion.TERRACE));
+		valueFunction = valueFunction.adaptWeight(Criterion.TERRACE, Criterion.TELE);
+		assertEquals(9d, valueFunction.getSubjectiveValueWeight(Criterion.TERRACE));
+		assertEquals(1d, valueFunction.getSubjectiveValueWeight(Criterion.TELE));
+
+	}
+
+	/**
+	 * Function to test if the bounds of an interval adapt well when needed
+	 */
+	@Test
+	void adaptBoundsTest() {
+
+		assertThrows(IllegalArgumentException.class, () -> valueFunction.adaptBounds(Criterion.TELE, 0d, true));
+		valueFunction.adaptBounds(Criterion.FLOOR_AREA_TERRACE, 25d, true);
+		LinearValueFunction lvf = (LinearValueFunction) valueFunction.getFloorAreaTerraceValueFunction();
+		assertEquals(25d, lvf.getInterval().lowerEndpoint());
+		assertEquals(0.6, valueFunction.getFloorAreaTerraceValueFunction().getSubjectiveValue(a.getFloorAreaTerrace()));
+		valueFunction.adaptBounds(Criterion.NB_BEDROOMS, 8, false);
+		double nbRooms = a.getNbBedrooms();
+		assertEquals(0d, valueFunction.getNbBedroomsValueFunction().getSubjectiveValue(nbRooms));
+		lvf = (LinearValueFunction) valueFunction.getNbBedroomsValueFunction();
+		assertEquals(8, lvf.getInterval().upperEndpoint());
+	}
+
+	/**
+	 * Function to test if a random apartment generated respects some criteria
+	 */
+	@Test
+	void getRandomApartmentValueFunctionTest() {
+
+		ApartmentValueFunction apart = ApartmentValueFunction.getRandomApartmentValueFunction();
+		assertEquals(1d, apart.getFloorAreaValueFunction().getSubjectiveValue(a.getFloorArea()));
+		LinearValueFunction lvf = (LinearValueFunction) apart.getFloorAreaTerraceValueFunction();
+		assertEquals(true, lvf.getInterval().upperEndpoint() <= 101d);
```

Comment:
Mes rétines me brûlent. `assertTrue`, par pitié.

Reply:
Fixed in commit 706ab00